packages/opentelemetry-plugin-user-interaction/src/userInteraction.ts

```diff
@@ -0,0 +1,457 @@
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as shimmer from 'shimmer';
+import { BasePlugin, hrTime, isWrapped } from '@opentelemetry/core';
+import * as types from '@opentelemetry/types';
+import { getElementXPath } from '@opentelemetry/web';
+import {
+  AsyncTask,
+  RunTaskFunction,
+  SpanData,
+  WindowWithZone,
+  ZoneTypeWithPrototype,
+} from './types';
+import { AttributeNames } from './enums/AttributeNames';
+import { VERSION } from './version';
+
+const ZONE_SCOPE_KEY = 'OT_ZONE_SCOPE';
+const EVENT_CLICK_NAME = 'event_click:';
+const EVENT_NAVIGATION_NAME = 'Navigation:';
+
+/**
+ * This class represents a UserInteraction plugin for auto instrumentation.
+ * If zone.js is available then it patches the zone otherwise it patches
+ * addEventListener of HTMLElement
+ */
+export class UserInteractionPlugin extends BasePlugin<unknown> {
+  readonly component: string = 'user-interaction';
+  readonly version = VERSION;
+  moduleName = this.component;
+  private _spansData = new WeakMap<types.Span, SpanData>();
+  private _zonePatched = false;
+
+  constructor() {
+    super('@opentelemetry/plugin-user-interaction', VERSION);
+  }
+
+  /**
+   * This will check if last task was timeout and will save the time to
+   * fix the user interaction when nothing happens
+   * This timeout comes from xhr plugin which is needed to collect information
+   * about last xhr main request from observer
+   * @param task
+   * @param span
+   */
+  private _checkForTimeout(task: AsyncTask, span: types.Span) {
+    const spanData = this._spansData.get(span);
+    if (spanData) {
+      if (task.source === 'setTimeout') {
+        spanData.hrTimeLastTimeout = hrTime();
+      } else if (
+        task.source !== 'Promise.then' &&
+        task.source !== 'setTimeout'
+      ) {
+        spanData.hrTimeLastTimeout = undefined;
+      }
+    }
+  }
+
+  /**
+   * Creates a new span
+   * @param element
+   * @param eventName
+   */
+  private _createSpan(
+    element: HTMLElement,
+    eventName: string
+  ): types.Span | undefined {
+    if (!element.getAttribute) {
+      return undefined;
+    }
+    if (element.hasAttribute('disabled')) {
+      return undefined;
+    }
+    const xpath = getElementXPath(element, true);
+    try {
+      const span = this._tracer.startSpan(`${EVENT_CLICK_NAME} ${xpath}`, {
+        attributes: {
+          [AttributeNames.COMPONENT]: this.component,
+          [AttributeNames.EVENT_TYPE]: eventName,
+          [AttributeNames.TARGET_ELEMENT]: element.tagName,
+          [AttributeNames.TARGET_XPATH]: xpath,
+          [AttributeNames.HTTP_URL]: window.location.href,
```

Comment:
Is this meant to be the page you're currently viewing when you perform an interaction? If yes, I think we should use a different attribute name. `http` is a wire protocol and may not accurately represent the intended purpose. In addition, `http.url` already has a meaning in the spec and I don't want to confuse people. Can we change this to be something like `view.url` or `page.url` or similar?

@mayurkale22 @open-telemetry/javascript-approvers WDYT?

Reply:
One of the complexities here is that the user did in fact use the HTTP protocol to request the page initially. In that sense, it's sort of tied back to the original request. It also may be what people are most familiar with searching for if they were to think of it.

That said, it is true that `view.url` or `page.url` is more accurate. If we were to do with that, we should also update the Document load plugin to reference it in its root spans (see https://github.com/open-telemetry/opentelemetry-js/blob/d3af8c4e0b5cbed8388b87f710989d7d5848eb07/packages/opentelemetry-plugin-document-load/src/documentLoad.ts#L117)

Reply:
@obecny @mayurkale22 what are your thoughts here?

Reply:
I don't have any strong opinion on that, for me both 3 options are fine but having the same name for url if this is either web or node makes more sense for me so that's why I added here the same name as in other plugins. But this is really a minor thing and can be changed at anytime now or later.